src/main/java/org/spongepowered/api/data/manipulator/immutable/item/ImmutableMapItemData.java

```diff
@@ -24,15 +24,63 @@
  */
 package org.spongepowered.api.data.manipulator.immutable.item;
 
+import com.flowpowered.math.vector.Vector2i;
 import org.spongepowered.api.data.manipulator.ImmutableDataManipulator;
 import org.spongepowered.api.data.manipulator.mutable.item.MapItemData;
+import org.spongepowered.api.data.value.immutable.ImmutableBoundedValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
 import org.spongepowered.api.item.ItemTypes;
 import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.map.MapCanvas;
+import org.spongepowered.api.world.World;
 
 /**
  * Represents an {@link ImmutableDataManipulator} hosting the specific map
  * information of an {@link ItemStack} of the type {@link ItemTypes#FILLED_MAP}.
+ * Warning: This will affect all maps with this id (damage value).
```

Comment:
There's no understanding of damage value or id in the API, do you mean the map data itself? There's no possible way to alter an immutable data once it's created, so it shouldn't be altered by any existing map.